app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -0,0 +1,487 @@
+            }
+            else -> {
+                articleAndImageContributions.addAll(articleContributions)
+                articleAndImageContributions.addAll(imageContributions)
+            }
+        }
+    }
+
+    private class HeaderViewHolder internal constructor(itemView: View) : DefaultViewHolder<View?>(itemView) {
+        var headerText: TextView = itemView.findViewById(R.id.section_header_text)
+        fun bindItem(date: String) {
+            headerText.text = date
+            headerText.setTextColor(ResourceUtil.getThemedColor(headerText.context, R.attr.colorAccent))
+        }
+    }
+
+    private class ContributionItemHolder internal constructor(itemView: SuggestedEditsContributionsItemView) : DefaultViewHolder<SuggestedEditsContributionsItemView?>(itemView) {
+        val disposables = CompositeDisposable()
+        fun bindItem(contribution: Contribution) {
+            view.setTitle(contribution.description)
+            view.setDescription(contribution.title)
+            view.setIcon(contribution.editType)
+            view.setImageUrl(contribution.imageUrl)
+            getPageView(view, contribution)
+            getImageDetails(view, contribution)
+        }
+
+        fun clearDisposables() {
+            disposables.clear()
+        }
+
+        private fun getImageDetails(itemView: SuggestedEditsContributionsItemView, contribution: Contribution) {
+            if (contribution.editType == EDIT_TYPE_ARTICLE_DESCRIPTION) {
+                disposables.add(ServiceFactory.getRest(contribution.wikiSite).getSummary(null, contribution.title)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe({ summary ->
+                            contribution.imageUrl = summary.thumbnailUrl.toString()
+                            itemView.setImageUrl(contribution.imageUrl)
+                        }, { t ->
+                            L.e(t)
+                        }))
+            } else {
+                disposables.add(ServiceFactory.get(WikiSite(Service.COMMONS_URL)).getImageInfo(contribution.title, contribution.wikiSite.languageCode())
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe({ response ->
+                            val page = response.query()!!.pages()!![0]
+                            val labels = ArrayList<String>()
+                            if (page.imageInfo() != null) {
+                                val imageInfo = page.imageInfo()!!
+                                contribution.description = imageInfo.metadata!!.imageDescription()
+                                contribution.imageUrl = imageInfo.originalUrl
+                                if (contribution.editType == EDIT_TYPE_IMAGE_TAG) {
+                                    if (!page.imageLabels.isNullOrEmpty()) {
```

Comment:
"image labels" are not the same thing as image tags, and do not belong here.  This can be done in a future PR, but if we want to show image tags for the current image, it will need to be another API call.